client/finality-grandpa/src/finality_proof.rs

```diff
@@ -145,6 +147,55 @@ impl<Block: BlockT> AuthoritySetForFinalityChecker<Block> for Arc<dyn FetchCheck
 	}
 }
 
+
+/// Justification for a finalized block.
+#[derive(Clone)]
+pub struct JustificationNotification<Block: BlockT> {
+	/// Highest finalized block header
+	pub header: Block::Header,
+	/// An encoded justification proving that the given header has been finalized
+	pub justification: Vec<u8>,
+}
+
+type JustificationStream<Block> = TracingUnboundedReceiver<JustificationNotification<Block>>;
+pub type SharedFinalitySubscribers<T> = Arc<RwLock<Vec<JustificationStream<T>>>>;
```

Comment:
Why is that ever needed? Each subscriber should manage it's own stream - and most likely what they do is they consume it, convert into a `Future` and spawn on the runtime.